gdal/ogr/ogrsf_frmts/elastic/ogrelasticlayer.cpp

```diff
@@ -529,9 +532,12 @@ void OGRElasticLayer::FinalizeFeatureDefn(bool bReadFeatures)
                     osPostData = m_osESSearch;
                 }
                 else
-                    osRequest = CPLSPrintf("%s/%s/%s/_search?scroll=1m&size=%d",
-                           m_poDS->GetURL(), m_osIndexName.c_str(),
-                           m_osMappingName.c_str(), m_poDS->m_nBatchSize);
+                {
+                    osRequest = CPLSPrintf("%s/%s", m_poDS->GetURL(), m_osIndexName.c_str());
```

Comment:
to avoid most of the conditional tests of version < 7 in that file, shouldn't we have a method BuildFullURL() (or a better name) that would return "GetURL()/m_osIndexName/m_osMappingName" for < 7 or ""GetURL()/m_osIndexName" for >= 7 ?